src/Microsoft.IdentityModel.Tokens/Validators.cs

```diff
@@ -207,16 +208,19 @@ public static void ValidateIssuerSecurityKey(SecurityKey securityKey, SecurityTo
                 throw LogHelper.LogArgumentNullException(nameof(securityToken));
 
             X509SecurityKey x509SecurityKey = securityKey as X509SecurityKey;
-            if (x509SecurityKey != null)
+            if (x509SecurityKey?.Certificate is X509Certificate2 cert)
             {
-                var cert = x509SecurityKey.Certificate;
-                DateTime utcNow = DateTime.UtcNow;
+                DateTime localTime = DateTime.Now;
```

Comment:
The root cause of this issue is a comparison between local and universal time. Rather than making these comparisons all local, have you considered making them all universal? This would keep the code robust against various time concerns, such as differences in time zones, daylight saving, etc. I would recommend bringing back the original implementation with an update to call the ToUniversalTime() method on the NotBefore property of the certificate and comparing that value against UtcNow.

Reply:
thanks Chris.
we found out that X509Certificate2 returns localtime rather than utc time.
we don't have any control over that.

Reply:
It can be easily converted into a UTC kind of DateTime using the DateTime.ToUniversalTime() method. For example: cert.NotBefore.ToUniversalTime()

Reply:
Just made this change. @CBaud Thanks for explaining why this is a good idea!